tavern_pathfinder/src/lib.rs

```diff
@@ -1,7 +1,622 @@
+    	KnowledgeNobility, 
+    	KnowledgePlanes, 
+    	KnowledgeReligion, 
+    	Linguistics, 
+    	Perception, 
+    	Perform, 
+    	Profession, 
+    	Ride, 
+    	SenseMotive, 
+    	SleightOfHand, 
+    	Spellcraft,
+        Stealth,
+        Survival,
+        Swim,
+        UseMagicDevice,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum SaveThrow {
+        Fortitude,
+        Reflex,
+        Will,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum CharacterStat {
+    	Name,
+    	Race,
+    	Size,
+    	Height,
+    	Weight,
+    	Age,
+    	Gender,
+    	Alignment,
+    	Deity,
+    	Languages,
+        Appearance,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum CombatStat {
+    	MeleeAttackBonus,
+    	RangedAttackBonus,
+    	CMB,
+    	CMD,
+    	ArmorClass,
+    	TouchAC,
+    	FlatFootedAC,
+    	InitiativeBonus,
+    	DamageReduction,
+    	SpellResistance,
+        Speed,
+        FortitudeSave,
+        ReflexSave,
+        WillSave,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum CasterType {
+        Spontaneous,
+        Prepared,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum ComponentType {
+        Somatic,
+        Material,
+        Verbal,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum MagicSchool {    
+    	Abjuration, 
+    	Conjuration, 
+    	Divination, 
+    	Enchantment, 
+    	Evocation, 
+    	Illusion, 
+    	Necromancy, 
+	    Transmutation,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum SpellRange {
+        Personal,
+        Touch,
+        Close,
+        Medium,
+        Long,
+        Unlimited,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum WeaponClass {
+    	Axes, 
+    	HeavyBlades, 
+    	LightBlades, 
+    	Bows, 
+    	Close, 
+    	Crossbows, 
+    	Double, 
+    	Firearms, 
+    	Flails, 
+    	Hammers, 
+    	Monk, 
+    	Natural, 
+    	Polearms, 
+	    SiegeEngines, 
+	    Spears,
+        Thrown,
+        Tribal,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum ArmorClass {
+        Light,
+        Medium,
+        Heavy,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum EquipmentSlot {
+    	NoSlot,
+    	Armor, 
+    	Belts, 
+    	Body, 
+    	Chest, 
+    	Eyes, 
+    	Feet, 
+    	Hands, 
+    	Head, 
+    	Headband, 
+    	Neck, 
+    	RingLeft, 
+    	RingRight, 
+    	Shield, 
+    	Shoulders,
+        Wrist,
+    }
+
+    #[derive(Serialize,Deserialize)]
+    pub enum DamageType {
+    	Bludgeoning, 
+    	Slashing, 
+    	Piercing, 
+    	Energy, 
+    	Acid, 
+       	Fire, 
+    	Electricity, 
+    	Cold, 
+    	Sonic, 
+    	Positive, 
+    	Negative, 
+	    Nonlethal,
+    }
+}
+
+pub mod entities {
+    pub mod character {
+        use serde::{Serialize,Deserialize};
+        use crate::types;
+        use uuid::Uuid;
+
+        #[derive(Serialize, Deserialize)]
+        pub struct Character {
+            id: Uuid,
+            char_id: Uuid,
```

Comment:
What is the difference between `id` and `char_id`, and can we make this more clear --- keeping in mind my other note about changing some `Uuid` fields to the actual type or a `Summary` variant.

Reply:
This field got dropped during the first revision - it was a copy of the _user_ id from the database specs.